object-cache.php

```diff
@@ -1148,13 +1148,20 @@ public function __construct() {
 				$l1 = new APCuL1();
 			}
 
-			list( $port, $socket ) = self::get_port_socket_from_host( DB_HOST );
+			list( $port, $socket, $host ) = self::get_port_socket_from_host( DB_HOST );
 
 			if ( defined( 'WP_LCACHE_RUNNING_TESTS' ) && WP_LCACHE_RUNNING_TESTS ) {
 				wp_lcache_initialize_database_schema();
 			}
 
-			$dsn = 'mysql:host='. DB_HOST. ';port='. $port .';dbname='. DB_NAME;
+			if ( $socket ) {
+				$dsn = sprintf( 'mysql:unix_socket=%s;dbname=%s;charset=%s', DB_HOST, DB_NAME, DB_CHARSET );
+			} elseif ( $port ) {
+				$dsn = sprintf( 'mysql:host=%s;port=%d;dbname=%s;charset=%s', $host, $port, DB_NAME, DB_CHARSET );
+			} else {
+				$dsn = sprintf( 'mysql:host=%s;dbname=%s;charset=%s', DB_HOST, DB_NAME, DB_CHARSET );
+			}
```

Comment:
@danielbachhuber Done in https://github.com/lcache/wp-lcache/pull/115/commits/b5da7055c4a1ec8cd52e768fe43335fb6dda8bcd